tests/phpunit/FileSystem/FileSystemTestCase.php

```diff
@@ -35,62 +35,42 @@
 
 namespace Infection\Tests\FileSystem;
 
-use function getenv;
-use function Infection\Tests\make_tmp_dir;
+use Infection\Tests\TestingUtility\FS;
 use PHPUnit\Framework\TestCase;
 use function Safe\getcwd;
 use function Safe\realpath;
-use function sprintf;
+use function str_replace;
 use Symfony\Component\Filesystem\Filesystem;
-use Symfony\Component\Filesystem\Path;
-use function sys_get_temp_dir;
 
 /**
  * @private
  */
 abstract class FileSystemTestCase extends TestCase
 {
-    private const TMP_DIR_NAME = 'infection-test';
-
     protected string $cwd = '';
 
     protected string $tmp = '';
 
-    public static function tearDownAfterClass(): void
-    {
-        // Cleans up whatever was there before. Indeed upon failure PHPUnit fails to trigger the
-        // `tearDown()` method and as a result some temporary files may still remain.
-        self::removeTmpDir();
-    }
-
     protected function setUp(): void
     {
-        // Cleans up whatever was there before. Indeed upon failure PHPUnit fails to trigger the
-        // `tearDown()` method and as a result some temporary files may still remain.
-        self::removeTmpDir();
-
         $this->cwd = getcwd();
-        $this->tmp = make_tmp_dir(self::TMP_DIR_NAME, self::class);
+        $this->tmp = realpath(
+            FS::tmpDir(
+                $this->getTmpDirPrefix(),
+            ),
+        );
     }
 
     protected function tearDown(): void
     {
         (new Filesystem())->remove($this->tmp);
     }
 
-    final protected static function removeTmpDir(): void
+    /**
+     * If the test case is `App\Tests\MyFilesystemServiceTestCase`, the default prefix will be "App\Tests\MyFilesystemServiceTestCase".
+     */
+    protected function getTmpDirPrefix(): string
     {
-        $testToken = getenv('TEST_TOKEN');
```

Comment:
I'm pretty sure this was added to work with paratest (its support though has been dropped).

Will it work now with parallel execution?

Reply:
Since every directory created is unique, there is no need to add a `TEST_TOKEN` value. It was only relevant before because for one process, we were using the same root directory which gets cleaned up as well by that process. So if you had 2 processes at the same time, one could have deleted that parent directory while the other was still going on.

As a downside the temporary directories are "less contained", as they are no longer in a `/path/to/sys-tmp-dir/Infection` directory, but they will still look like `/path/to/sys-tmp-dir/InfectionTests...ClassName8d9e0f1a` so you can always easily do `rm -r /path/to/sys-tmp-dir/Infection*`